src/tsn_adapters/flows/eps/historical_flow.py

```
@@ -0,0 +1,163 @@
+
+
+@task
+def read_published_dates(tn_block: TNAccessBlock, stream_id: str) -> set[int]:
+    """Return unix timestamps already in TN for the given stream."""
+    try:
+        df = tn_block.read_records(
+            stream_id=stream_id,
+            date_from=date_string_to_unix(BACKFILL_START),
+        )
+        return set(df["date"].tolist())
+    except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)